src/neurostatslib/utils.py

```diff
@@ -383,3 +383,34 @@ def row_wise_kron(a, c):
         K = K.T
 
     return K
+
+
+def has_local_device(device_type: str) -> bool:
+    """Scan for local device availability.
+
+    Looks for local device availability and returns True if the specified
+    type of device (e.g., GPU, TPU) is available.
+
+    Parameters
+    ----------
+    device_type:
+        The the device type in lower-case, e.g. `gpu`, `tpu`...
+
+    Returns
+    -------
+    :
+        True if the jax finds the device, False otherwise.
+
+    """
+    return any(
+        device_type in device.device_kind.lower() for device in jax.local_devices()
+    )
+
+
+def is_sequence(obj) -> bool:
+    """Check if an object is a sequence."""
```

Comment:
feel like this docstring / title should be updated: str, bytes, dict are all sequences. what you're looking for is something list-like, right?

Reply:
I changed the description and name, btw is dict is a sequence?